### ISSUE TITLE
Don't initialize variables to null

### ISSUE THREAD

**Issue by hterkelsen** [dart-lang/linter#160]
This is not allowed according to Effective Dart:
https://www.dartlang.org/effective-dart/usage/#dont-explicitly-initialize-variables-to-null

BAD:
```dart
var foo = null;
```

GOOD:
```dart
var foo;
```

**Comment by pq** [comment]
Nice one!

https://codereview.chromium.org/1633303002/

**Comment by pq** [comment]
Implemented w/ f9fa5f55fda0d6263ffc0effc52168bee1b8ed31

**Comment by a9udn9u** [comment]
How about `final` variables? For instance:

```dart
class Foo {
  final dynamic bar;
  Foo({ this.bar = null });
}
```

I want to assign `null` to `bar` if the user omits it in the constructor. If not assign `null` then `bar` becomes unassigned.

**Comment by bwilkerson** [comment]
Whether final or not, all variables are automatically initialized to `null` if there is no explicit initializer. The two parameter declarations `{this.bar}` and `{this.bar = null}` have exactly the same run-time behavior. In both cases the field `bar` will be initialized to `null`. There is no concept in Dart of "unassigned". You can see this for yourself by executing the following:
```
class Foo {
  final dynamic bar;
  Foo({this.bar});
}
main() {
  print(Foo().bar);
}
```
